fabric-renderer-indigo/src/main/java/net/fabricmc/indigo/renderer/mixin/MixinBufferBuilder.java

```diff
@@ -36,21 +36,22 @@
     @Shadow abstract int getCurrentSize();
     @Shadow public abstract VertexFormat getVertexFormat();
 
-	private static final int VERTEX_STRIDE_INTS = 7;
+    private static final int VERTEX_STRIDE_INTS = 8;
     private static final int QUAD_STRIDE_INTS = VERTEX_STRIDE_INTS * 4;
     private static final int QUAD_STRIDE_BYTES = QUAD_STRIDE_INTS * 4;
 
     @Override
     public void fabric_putQuad(QuadViewImpl quad) {
-        if(Indigo.ENSURE_VERTEX_FORMAT_COMPATIBILITY) {
+        if (Indigo.ENSURE_VERTEX_FORMAT_COMPATIBILITY) {
             bufferCompatibly(quad);
         } else {
             bufferFast(quad);
         }
     }
 
     private void bufferFast(QuadViewImpl quad) {
-        grow(QUAD_STRIDE_BYTES);
+        grow(QUAD_STRIDE_BYTES + getVertexFormat().getVertexSize());
+        bufInt.limit(bufInt.capacity());
```

Comment:
I'd expect grow to adjust the limit, not sure why you grow more/differently than before?

Reply:
Cuz it broke. :-)

Not sure why either, but this is exactly what vanilla does, so I did the same thing in the interest of time and the overflow error stopped happening.  Is a puzzle, tho.